scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamOp.scala

```diff
@@ -59,6 +61,25 @@ object BeamOp extends Serializable {
   )(implicit ordK: Ordering[K], kryoCoder: KryoCoder): PCollection[KV[K, java.lang.Iterable[U]]] = {
     reduceFn match {
       case ComposedMapGroup(f, g) => planMapGroup(planMapGroup(pcoll, f), g)
+      case EmptyGuard(MapValueStream(SumAll(pqm: PriorityQueueMonoid[V]))) =>
```

Comment:
I don't *think* this should either compile or run correctly. I probably wrote some bad code in my example. The `SumAll` type is `TraversableOnce[A] => Iterator[A]` where in this case `A` = `V`. So when you pattern match on `PriorityQueueMonoid[T]` that means `V = PriorityQueue[T]`, so, it shouldn't be `PriorityQueueMonoid[V]`. A little known fact is that you can use a lower case to bind an unknown type: `pqm: PriorityQueueMonoid[v]` and then use `v` lower down.